src/goModifytags.ts

```diff
@@ -0,0 +1,140 @@
+		if (!tags && !options) {
+			return;
+		}
+		if (tags) {
+			args.push('--add-tags');
+			args.push(tags);
+		}
+		if (options) {
+			args.push('--add-options');
+			args.push(options);
+		}
+		runGomodifytags(args);
+	});
+
+}
+
+export function removeTags(commandArgs: GoTagsConfig) {
+	let args = getCommonArgs();
+	if (!args) {
+		return;
+	}
+
+	getTagsAndOptions(<GoTagsConfig>vscode.workspace.getConfiguration('go')['removeTags'], commandArgs).then(([tags, options]) => {
+		if (!tags && !options) {
+			args.push('--clear-tags');
+			args.push('--clear-options');
+		}
+		if (tags) {
+			args.push('--remove-tags');
+			args.push(tags);
+		}
+		if (options) {
+			args.push('--remove-options');
+			args.push(options);
+		}
+		runGomodifytags(args);
+	});
+}
+
+function getCommonArgs(): string[] {
+	let editor = vscode.window.activeTextEditor;
+	if (!editor) {
+		vscode.window.showInformationMessage('No editor is active.');
+		return;
+	}
+	if (!editor.document.fileName.endsWith('.go')) {
+		vscode.window.showInformationMessage('Current file is not a Go file.');
+		return;
+	}
+	let args = ['-modified', '-format', 'json'];
+	if (editor.selection.start.line === editor.selection.end.line && editor.selection.start.character === editor.selection.end.character) {
+		// Add tags to the whole struct
+		let offset = byteOffsetAt(editor.document, editor.selection.start);
+		args.push('-offset');
+		args.push(offset.toString());
+	} else if (editor.selection.start.line <= editor.selection.end.line) {
+		// Add tags to selected lines
+		args.push('-line');
+		args.push(`${editor.selection.start.line + 1},${editor.selection.end.line + 1}`);
+	}
+
+	return args;
+}
+
+function getTagsAndOptions(config: GoTagsConfig, commandArgs: GoTagsConfig): Thenable<string[]> {
+	let tags = commandArgs && commandArgs.hasOwnProperty('tags') ? commandArgs['tags'] : config['tags'];
+	let options =  commandArgs && commandArgs.hasOwnProperty('options') ? commandArgs['options'] : config['options'];
+	let promptForTags =  commandArgs && commandArgs.hasOwnProperty('promptForTags') ? commandArgs['promptForTags'] : config['promptForTags'];
+
+	if (!promptForTags) {
+		return Promise.resolve([tags, options]);
+	}
+
+	return vscode.window.showInputBox({
+		value: 'json',
+		prompt: 'Enter comma separated tag names'
+	}).then(inputTags => {
+		return vscode.window.showInputBox({
+			value: 'json=omitempty,xml=cdata',
+			prompt: 'Enter comma separated options'
+		}).then(inputOptions => {
+			return [inputTags, inputOptions];
+		});
+	});
+}
+
+function runGomodifytags(args: string[]) {
+	let gomodifytags = getBinPath('gomodifytags');
+	let editor = vscode.window.activeTextEditor;
+	let fileContents = editor.document.getText();
+	let input = editor.document.fileName + '\n' + fileContents.length + '\n' + fileContents;
+	let p = cp.execFile(gomodifytags, args, (err, stdout, stderr) => {
+		if (err && (<any>err).code === 'ENOENT') {
+			promptForMissingTool('gomodifytags');
```

Comment:
We could, but this is the model we follow for all other tools, so will not be changing this unless we change the rest of the places